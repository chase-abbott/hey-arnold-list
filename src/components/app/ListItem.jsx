import React from 'react';
import styled from 'styled-components';
import PropTypes from 'prop-types';

const Image = styled.img`
`;

export default function ListItem({ image }){
  return <Image src={image} alt="image"/>;
}

ListItem.propTypes = {
  image: PropTypes.string.isRequired
};
