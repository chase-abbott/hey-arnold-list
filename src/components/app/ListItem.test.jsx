/* eslint-disable max-len */
import React from 'react';
import { render } from '@testing-library/react';
import ListItem from './ListItem';

describe('List item component', () => {
  it('renders a list item to the page', () => {
    const url = 'https://vignette.wikia.nocookie.net/heyarnold/images/4/42/Arnie.jpg/revision/latest/scale-to-width-down/310?cb=20110109195825';
    const { asFragment } = render(<ListItem image={url}/>);
    expect(asFragment()).toMatchSnapshot();
  });
});
