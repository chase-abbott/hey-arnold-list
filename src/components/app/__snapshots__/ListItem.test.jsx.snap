// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`List item component renders a list item to the page 1`] = `
<DocumentFragment>
  <img
    alt="image"
    class="sc-bdnxRM gAuIVV"
    src="https://vignette.wikia.nocookie.net/heyarnold/images/4/42/Arnie.jpg/revision/latest/scale-to-width-down/310?cb=20110109195825"
  />
</DocumentFragment>
`;
